From f5f299900f2247acc81f24554c9bfe6c8ae4d1d3 Mon Sep 17 00:00:00 2001
From: Eric Garver <eric@garver.life>
Date: Wed, 29 Oct 2025 11:51:18 -0400
Subject: [PATCH 35/37] v2.4.0: chore(nftables): move _icmp_types_fragments()
 inside the class

It was the only function outside of the class and is not useful to any
other code. Put it inside the class with everything else.

(cherry picked from commit 69ad16a4435a0b49b4196aa1a99ee963b72c4b69)
---
 src/firewall/core/nftables.py | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/src/firewall/core/nftables.py b/src/firewall/core/nftables.py
index ef059150917b..ba0aa7845838 100644
--- a/src/firewall/core/nftables.py
+++ b/src/firewall/core/nftables.py
@@ -81,16 +81,6 @@ IPTABLES_TO_NFT_HOOK = {
     },
 }
 
-def _icmp_types_fragments(protocol, type, code=None):
-    fragments = [{"match": {"left": {"payload": {"protocol": protocol, "field": "type"}},
-                            "op": "==",
-                            "right": type}}]
-    if code is not None:
-        fragments.append({"match": {"left": {"payload": {"protocol": protocol, "field": "code"}},
-                                    "op": "==",
-                                    "right": code}})
-    return fragments
-
 class nftables(object):
     name = "nftables"
     policies_supported = True
@@ -1502,13 +1492,23 @@ class nftables(object):
 
         return rules
 
+    def _icmp_types_fragments(self, protocol, type, code=None):
+        fragments = [{"match": {"left": {"payload": {"protocol": protocol, "field": "type"}},
+                                "op": "==",
+                                "right": type}}]
+        if code is not None:
+            fragments.append({"match": {"left": {"payload": {"protocol": protocol, "field": "code"}},
+                                        "op": "==",
+                                        "right": code}})
+        return fragments
+
     def _icmp_types_to_nft_fragments(self, ipv, icmp_type):
         if ipv == "ipv4" and icmp_type in ICMP_TYPES:
             _type, _code, _omit_code = ICMP_TYPES[icmp_type]
-            return _icmp_types_fragments("icmp", _type, None if _omit_code else _code)
+            return self._icmp_types_fragments("icmp", _type, None if _omit_code else _code)
         elif ipv == "ipv6" and icmp_type in ICMPV6_TYPES:
             _type, _code, _omit_code = ICMPV6_TYPES[icmp_type]
-            return _icmp_types_fragments("icmpv6", _type, None if _omit_code else _code)
+            return self._icmp_types_fragments("icmpv6", _type, None if _omit_code else _code)
         else:
             raise FirewallError(INVALID_ICMPTYPE,
                                 "ICMP type '%s' not supported by %s for %s" % (icmp_type, self.name, ipv))
-- 
2.52.0

